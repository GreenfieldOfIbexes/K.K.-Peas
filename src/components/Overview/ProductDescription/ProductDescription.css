.product-description {
	display: flex;
	justify-content: space-between;
	margin: 80px auto 0;
	width: 100%;
}
.product-description .separator {
	border: 1px solid black;
	height: 100%;
	width: 0;
	margin: 0 20px;
}
.product-description .description {
	text-align: right;
	width: 50%;
	max-width: 500px;
}
.product-description .features {
	width: 50%;
	max-width: 500px;
}
.product-description .feature {
	display: flex;
	align-items: center;
}
