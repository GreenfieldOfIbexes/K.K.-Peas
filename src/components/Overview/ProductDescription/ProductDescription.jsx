import React from "react";
import "./ProductDescription.css";

const ProductDescription = ({ description, features }) => (
	<div className="product-description">
		<div className="description">{description}</div>
		<div className="separator"></div>
		<div className="features">
			{features.map((feature, index) => {
				if (feature.value !== "null") {
					return (
						<div className="feature" key={index}>
							<i className="material-icons">check</i>
							<span style={{ color: "transparent" }}>_</span>
							{feature.feature}
							{": "}
							{feature.value}
						</div>
					);
				}
			})}
		</div>
	</div>
);

export default ProductDescription;
