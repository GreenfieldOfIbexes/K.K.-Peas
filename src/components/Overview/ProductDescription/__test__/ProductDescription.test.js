import React from "react";
import renderer from "react-test-renderer";
import ProductDescription from "../ProductDescription.jsx";
import { testProduct } from "../../../../../tests/testProducts";

it("renders correctly", () => {
	const renderedComponent = renderer
		.create(
			<ProductDescription
				description={testProduct.description}
				features={testProduct.features}
			/>,
		)
		.toJSON();
	expect(renderedComponent).toMatchSnapshot();
});
