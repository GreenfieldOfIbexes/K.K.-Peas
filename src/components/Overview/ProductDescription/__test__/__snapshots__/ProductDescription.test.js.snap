// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="product-description"
>
  <div
    className="description"
  >
    The So Fatigues will wake you up and fit you in. This high energy camo will have you blending in to even the wildest surroundings.
  </div>
  <div
    className="separator"
  />
  <div
    className="features"
  >
    <div
      className="feature"
    >
      <i
        className="material-icons"
      >
        check
      </i>
      <span
        style={
          Object {
            "color": "transparent",
          }
        }
      >
        _
      </span>
      Buttons
      : 
      Brass
    </div>
  </div>
</div>
`;
